Cabal/Distribution/Types/SourceRepo.hs

```diff
@@ -123,20 +124,27 @@ instance NFData RepoKind where rnf = genericRnf
 -- 'SourceRepo' depend on the type of repo. The tools and methods used to
 -- obtain and track the repo depend on the repo type.
 --
-data RepoType = Darcs | Git | SVN | CVS
-              | Mercurial | GnuArch | Bazaar | Monotone
+data KnownRepoType = Darcs | Git | SVN | CVS
+                   | Mercurial | GnuArch | Bazaar | Monotone
+  deriving (Eq, Generic, Ord, Read, Show, Typeable, Data)
+
+instance Binary KnownRepoType
+instance Structured KnownRepoType
+instance NFData KnownRepoType where rnf = genericRnf
+
```

Comment:
Also `Parsec` and `Pretty` instance. And use them to implement `Pretty RepoType`.